addons/l10n_in/models/account_invoice.py

```diff
@@ -69,25 +69,45 @@ def _compute_l10n_in_state_id(self):
             else:
                 move.l10n_in_state_id = False
 
-    @api.depends('l10n_in_state_id')
+    @api.depends('l10n_in_state_id', 'l10n_in_gst_treatment')
     def _compute_fiscal_position_id(self):
-        other_country = None
-        in_moves = self.filtered(lambda move: move.country_code == 'IN' and move.is_sale_document(include_receipts=True))
+        in_moves = self.filtered(lambda move: move.country_code == 'IN' and move.is_invoice(include_receipts=True))
+        computed_ids = []
         for move in in_moves:
-            state_id = move.l10n_in_state_id.id
-            country_id = move.l10n_in_state_id.country_id.id
-            if move.l10n_in_state_id.l10n_in_tin == '96':  # Foreign Country
-                other_country = other_country or self.env['res.country'].search([('code', '!=', 'IN')], limit=1)
-                country_id = other_country.id
-                state_id = False
-            virtual_partner = self.env['res.partner'].new({
-                'state_id': state_id,
-                'country_id': country_id,
-            })
-            move.fiscal_position_id = self.env['account.fiscal.position'].with_company(
-                move.company_id
-            )._get_fiscal_position(virtual_partner)
-        super(AccountMove, self - in_moves)._compute_fiscal_position_id()
+            if (
+                move.partner_shipping_id.property_account_position_id
+                or move.partner_id.property_account_position_id
+            ):
+                continue
+            dummy_state_id = False
+            # If move type is sale, compare place of supply with company state
+            if move.is_sale_document(include_receipts=True):
```

Comment:
```suggestion
            if move.is_inbound():
```

Reply:
Since we don't want the same with fiscal (as discussed with PO) this comment is not applied 😬